### ISSUE TITLE
Can't login

### ISSUE THREAD

**Issue by hellnight** [skilion/onedrive#196]
Hi,
I have an issue when I try to login with the last commit : 

```
Authorize this app visiting:

https://login.microsoftonline.com/common/oauth2/v2.0/authorize?client_id=22c49a0d-d21c-4792-aed1-8f163c982546&scope=files.readwrite%20files.readwrite.all%20offline_access&response_type=code&redirect_uri=https://login.microsoftonline.com/common/oauth2/nativeclient

Enter the response uri: https://login.microsoftonline.com/common/oauth2/nativeclient?code=xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx


std.json.JSONException@std/json.d(493): Key not found: parentReference
----------------
??:? pure @safe void std.exception.bailOut!(std.json.JSONException).bailOut(immutable(char)[], ulong, const(char[])) [0x652e31]
??:? pure @safe inout(std.json.JSONValue)* std.exception.enforce!(std.json.JSONException, inout(std.json.JSONValue)*).enforce(inout(std.json.JSONValue)*, lazy const(char)[], immutable(char)[], ulong) [0x652e96]
??:? std.json.JSONValue.opIndexinout(pure ref @safe inout(std.json.JSONValue) function(immutable(char)[])) [0x60ad81]
src/sync.d:187 void sync.SyncEngine.applyDifference(std.json.JSONValue) [0x5d9e3e]
src/sync.d:163 void sync.SyncEngine.applyDifferences() [0x5d9aa6]
src/main.d:169 void main.performSync(sync.SyncEngine) [0x5c7f6f]
src/main.d:99 _Dmain [0x5c7790]
??:? _D2rt6dmain211_d_run_mainUiPPaPUAAaZiZ6runAllMFZ9__lambda1MFNlZv [0x5f12ab]
??:? scope void rt.dmain2._d_run_main(int, char**, extern (C) int function(char[][])*).tryExec(scope void delegate()) [0x5f11db]
??:? scope void rt.dmain2._d_run_main(int, char**, extern (C) int function(char[][])*).runAll() [0x5f1254]
??:? scope void rt.dmain2._d_run_main(int, char**, extern (C) int function(char[][])*).tryExec(scope void delegate()) [0x5f11db]
??:? _d_run_main [0x5f114b]
??:? main [0x5cb94b]
??:? __libc_start_main [0xb859382f]
Erreur de segmentation (core dumped)
```
If I back to commit 74c931c, I can connect without error. I think there are a issue on the last commit.

Regard

**Comment by skilion** [comment]
Fixed in 126f2719b838ef3ffb085591bb9fff62bd17e4f8
Are you using OneDrive for Business?

**Comment by hellnight** [comment]
Yes I use Onedrive for Business

**Comment by skilion** [comment]
Good, the switch to the Microsoft Graph API paid off. I wasn't really planning to support OneDrive for Business but it came almost naturally.

**Comment by hellnight** [comment]
I have the same issue with the last commit @skilion.

**Comment by skilion** [comment]
Can you compile the app with `make debug`, then run `./onedrive -v --resync`, and post the output?

**Comment by hellnight** [comment]
```
Initializing the Synchronization Engine ...
Applying differences ...
--anonymize--
The item is already present
--anonymize--
The item is already present
--anonymize--
The item is already present
--anonymize--
std.json.JSONException@std/json.d(493): Key not found: file
----------------
??:? pure @safe void std.exception.bailOut!(std.json.JSONException).bailOut(immutable(char)[], ulong, const(char[])) [0x65f8b9]
??:? pure @safe inout(std.json.JSONValue)* std.exception.enforce!(std.json.JSONException, inout(std.json.JSONValue)*).enforce(inout(std.json.JSONValue)*, lazy const(char)[], immutable(char)[], ulong) [0x65f91e]
??:? std.json.JSONValue.opIndexinout(pure ref @safe inout(std.json.JSONValue) function(immutable(char)[])) [0x61c359]
src/sync.d:67 itemdb.Item sync.makeItem(ref const(std.json.JSONValue)) [0x5eed91]
src/sync.d:194 void sync.SyncEngine.applyDifference(std.json.JSONValue) [0x5ef9db]
src/sync.d:169 void sync.SyncEngine.applyDifferences() [0x5ef671]
src/main.d:169 void main.performSync(sync.SyncEngine) [0x5e0abb]
src/main.d:99 _Dmain [0x5e0671]
??:? _D2rt6dmain211_d_run_mainUiPPaPUAAaZiZ6runAllMFZ9__lambda1MFNlZv [0x605bf7]
??:? scope void rt.dmain2._d_run_main(int, char**, extern (C) int function(char[][])*).tryExec(scope void delegate()) [0x605b27]
??:? scope void rt.dmain2._d_run_main(int, char**, extern (C) int function(char[][])*).runAll() [0x605ba0]
??:? scope void rt.dmain2._d_run_main(int, char**, extern (C) int function(char[][])*).tryExec(scope void delegate()) [0x605b27]
??:? _d_run_main [0x605a97]
??:? main [0x5e4449]
??:? __libc_start_main [0xe68ae82f]
Erreur de segmentation (core dumped)
```

**Comment by skilion** [comment]
Thanks, fixed in 974be0884b62864d38897ccdac2a1e7f8fee7255

**Comment by hellnight** [comment]
Thanks, it's work now.
